language-plutus-core/generators/Language/PlutusCore/Generators/Internal/TypedBuiltinGen.hs

```diff
@@ -56,54 +58,61 @@ type TypedBuiltinGen = TypedBuiltinGenT Identity
 instance PrettyCfg a => PrettyCfg (TermOf a) where
     prettyCfg cfg (TermOf t x) = prettyCfg cfg t <+> "~>" <+> prettyCfg cfg x
 
-attachCoercedTerm :: Functor f => TypedBuiltin Size a -> GenT f a -> GenT f (TermOf a)
-attachCoercedTerm tb = fmap $ \x -> TermOf (unsafeMakeBuiltin $ TypedBuiltinValue tb x) x
+attachCoercedTerm :: MonadQuote m => TypedBuiltin Size a -> GenT m a -> GenT m (TermOf a)
+attachCoercedTerm tb genX = do
+    x <- genX
+    term <- lift . liftQuote . unsafeMakeBuiltin $ TypedBuiltinValue tb x
```

Comment:
`MonadQuote` has a nice default implementation so it's a one-line derivation for anything implementing `MonadTrans`. So you could write `instance MonadQuote m => MonadQuote (GenT m)` and it should just work.

Reply:
Only if you want though, this is also fine.